### ISSUE TITLE
4.7.1.2 - HTTP 404s loading font-awesome fonts

### ISSUE THREAD

**Issue by 4oo4** [firefly-iii/firefly-iii#1225]
I am running Firefly III version 4.7.1.2, just upgraded from 4.7.0 (also made sure to follow the new upgrade steps).

#### Description of my issue:
After upgrading, the layout was broken in Firefox because it didn't have the proper path for the Font Awesome fonts. Even though I could also see the 404s in Chrome dev tools, the layout didn't break in Chrome. I was able to fix it by changing these URLs in `/public/css/app.css` from this:

```
@font-face{font-family:FontAwesome;src:url(/fonts/vendor/font-awesome/fontawesome-webfont.eo
t?674f50d287a8c48dc19ba404d20fe713);src:url(/fonts/vendor/font-awesome/fontawesome-webfont.eot?
674f50d287a8c48dc19ba404d20fe713) format("embedded-opentype"),url(/fonts/vendor/font-awesome/fo
ntawesome-webfont.woff2?af7ae505a9eed503f8b8e6982036873e) format("woff2"),url(/fonts/vendor/fon
t-awesome/fontawesome-webfont.woff?fee66e712a8a08eef5805a46892932ad) format("woff"),url(/fonts/
vendor/font-awesome/fontawesome-webfont.ttf?b06871f281fee6b241d60582ae9369b9) format("truetype"),url(/fonts/vendor/font-awesome/fontawesome-webfont.svg?912ec66d7572ff821749319396470bde) format("svg");font-weight:400;font-style:normal}
```

To this:
```
@font-face{font-family:FontAwesome;src:url(../fonts/vendor/font-awesome/fontawesome-webfont.eo
t?674f50d287a8c48dc19ba404d20fe713);src:url(../fonts/vendor/font-awesome/fontawesome-webfont.eot?
674f50d287a8c48dc19ba404d20fe713) format("embedded-opentype"),url(../fonts/vendor/font-awesome/fo
ntawesome-webfont.woff2?af7ae505a9eed503f8b8e6982036873e) format("woff2"),url(../fonts/vendor/fon
t-awesome/fontawesome-webfont.woff?fee66e712a8a08eef5805a46892932ad) format("woff"),url(../fonts/
vendor/font-awesome/fontawesome-webfont.ttf?b06871f281fee6b241d60582ae9369b9) format("truetype"),url(../fonts/vendor/font-awesome/fontawesome-webfont.svg?912ec66d7572ff821749319396470bde) format("svg");font-weight:400;font-style:normal}
```

I was going to try and submit a PR, but I wasn't sure where that part of `app.css` got pulled in from (perhaps one of the vendor packages?)

#### Steps to reproduce 
1) Upgrade Firefly
2) Login using Firefox

#### Other important details (log files, system info):
Browser console:

```
home:1187 GET https://example.com/fonts/vendor/font-awesome/fontawesome-webfont.woff2?af7ae505a9eed503f8b8e6982036873e net::ERR_ABORTED
home:1189 GET https://example.com/fonts/vendor/font-awesome/fontawesome-webfont.woff?fee66e712a8a08eef5805a46892932ad net::ERR_ABORTED
home:1189 GET https://example.com/fonts/vendor/font-awesome/fontawesome-webfont.ttf?b06871f281fee6b241d60582ae9369b9 net::ERR_ABORTED
```
My Firefly URL is `https://example.com/firefly/public`, so for me the correct URL would need to be `https://example.com/firefly/public/fonts/vendor/font-awesome`.

Cheers

**Comment by JC5** [comment]
I can see it in the source code but it shouldn't matter. There's a base href that should catch that. 

I'm trying to set up my own Firefly III in a sub dir to see what happens. I strongly recommend changing the Firefly III URL however

**Comment by 4oo4** [comment]
@JC5 Sounds good, how would you recommend changing the Firefly URL? I have my webserver set up to point to `/firefly/public` (to avoid security issues), is there something else about that subdirectory path I have that would mess with assets?

**Comment by JC5** [comment]
No, and that should already solve the problem you have. I'm trying to update the script so it will render "./" instead of "/" but no luck so far.

**Comment by Nik-vr** [comment]
I have a similar problem.

![1](https://user-images.githubusercontent.com/520760/36959830-3428c43e-2055-11e8-8b86-60c9ea6d2bb2.jpg)

**Comment by JC5** [comment]
If you run Firefly III in /some/directory/ instead of / will run into this problem unfortunately. I have no solution yet.

**Comment by Nik-vr** [comment]
I run Firefly from / on my server.

Start URL: http://some.domain.ru/public/

Errors: 
```
GET http://some.domain.ru/fonts/vendor/font-awesome/fontawesome-webfont.woff2?af7ae505a9eed503f8b8e6982036873e 404 (Not Found)
asset:1 GET http://some.domain.ru/fonts/vendor/font-awesome/fontawesome-webfont.woff?fee66e712a8a08eef5805a46892932ad net::ERR_ABORTED
asset:1 GET http://some.domain.ru/fonts/vendor/font-awesome/fontawesome-webfont.ttf?b06871f281fee6b241d60582ae9369b9 net::ERR_ABORTED
```

**Comment by Nik-vr** [comment]
On the local Windows server (which I use to update FireFly through the console) the way is:

http://fin.loc:8080/firefly-iii/public/

Errors:
```
GET http://fin.loc:8080/firefly-iii/public/jscript/variables?ext=.js&v=4.7.1.3 net::ERR_ABORTED
Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/fonts/vendor/font-awesome/fontawesome-webfont.woff2?af7ae505a9eed503f8b8e6982036873e
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Regular-latin.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-SemiBold-cyrillic.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-SemiBold-latin.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Light-latin.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Italic-latin.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Bold-cyrillic.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Bold-latin.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Light-cyrillic.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Regular-cyrillic.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Regular-latin-ext.woff2
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Italic-cyrillic.woff2
(index):1000 GET http://fin.loc:8080/fonts/vendor/font-awesome/fontawesome-webfont.woff2?af7ae505a9eed503f8b8e6982036873e net::ERR_ABORTED
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/fonts/vendor/font-awesome/fontawesome-webfont.woff?fee66e712a8a08eef5805a46892932ad
(index):1 Failed to decode downloaded font: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Italic-cyrillic.woff2
(index):1 OTS parsing error: cmap: No subtables in cmap!
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Italic-cyrillic.woff
(index):1000 GET http://fin.loc:8080/fonts/vendor/font-awesome/fontawesome-webfont.woff?fee66e712a8a08eef5805a46892932ad net::ERR_ABORTED
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/fonts/vendor/font-awesome/fontawesome-webfont.ttf?b06871f281fee6b241d60582ae9369b9
(index):1000 GET http://fin.loc:8080/fonts/vendor/font-awesome/fontawesome-webfont.ttf?b06871f281fee6b241d60582ae9369b9 net::ERR_ABORTED
(index):1000 GET http://fin.loc:8080/firefly-iii/public/jscript/variables?ext=.js&v=4.7.1.3 500 (Internal Server Error)
app.js?v=4.7.1.3:1 GET http://fin.loc:8080/firefly-iii/public/chart/account/frontpage 500 (Internal Server Error)
send @ app.js?v=4.7.1.3:1
ajax @ app.js?v=4.7.1.3:1
C.(anonymous function) @ app.js?v=4.7.1.3:1
getJSON @ app.js?v=4.7.1.3:1
drawAChart @ charts.js?v=4.7.1.3:312
lineChartWithDay @ charts.js?v=4.7.1.3:144
drawChart @ index.js?v=4.7.1.3:33
(anonymous) @ index.js?v=4.7.1.3:26
c @ app.js?v=4.7.1.3:1
l @ app.js?v=4.7.1.3:1
setTimeout (async)
(anonymous) @ app.js?v=4.7.1.3:1
u @ app.js?v=4.7.1.3:1
fireWith @ app.js?v=4.7.1.3:1
fire @ app.js?v=4.7.1.3:1
u @ app.js?v=4.7.1.3:1
fireWith @ app.js?v=4.7.1.3:1
ready @ app.js?v=4.7.1.3:1
W @ app.js?v=4.7.1.3:1
error.png:1 GET http://fin.loc:8080/images/error.png 404 (Not Found)
(index):1 Slow network is detected. Fallback font will be used while loading: http://fin.loc:8080/firefly-iii/public/fonts/SourceSansPro-Bold-latin-ext.woff2
app.js?v=4.7.1.3:1 GET http://fin.loc:8080/firefly-iii/public/json/box/net-worth 500 (Internal Server Error)
send @ app.js?v=4.7.1.3:1
ajax @ app.js?v=4.7.1.3:1
C.(anonymous function) @ app.js?v=4.7.1.3:1
getJSON @ app.js?v=4.7.1.3:1
getNetWorthBox @ index.js?v=4.7.1.3:69
drawChart @ index.js?v=4.7.1.3:50
(anonymous) @ index.js?v=4.7.1.3:26
c @ app.js?v=4.7.1.3:1
l @ app.js?v=4.7.1.3:1
setTimeout (async)
(anonymous) @ app.js?v=4.7.1.3:1
u @ app.js?v=4.7.1.3:1
fireWith @ app.js?v=4.7.1.3:1
fire @ app.js?v=4.7.1.3:1
u @ app.js?v=4.7.1.3:1
fireWith @ app.js?v=4.7.1.3:1
ready @ app.js?v=4.7.1.3:1
W @ app.js?v=4.7.1.3:1
app.js?v=4.7.1.3:1 GET http://fin.loc:8080/firefly-iii/public/json/box/balance 500 (Internal Server Error)
```

**Comment by Nik-vr** [comment]
In addition to the problem with the font, the diagrams do not work, and the symbol "<?>" in the quality of the bit separator is displayed.

![2](https://user-images.githubusercontent.com/520760/36960459-2ba5d790-2058-11e8-8cec-1c1a3adee6af.jpg)

**Comment by JC5** [comment]
I see some 500 errors in your logs. Could you share those?

**Comment by JC5** [comment]
The log in /storage/logs when you visit the index page.

**Comment by Nik-vr** [comment]
I delete old log-file and go to index page. New log:

[ff3-apache2handler-2018-03-05.log](https://github.com/firefly-iii/firefly-iii/files/1780185/ff3-apache2handler-2018-03-05.log)

**Comment by jinformatique** [comment]
Hi, me too I have problems with icons and picto. I just upgraded to 4.7.1.3. Nothing changed in my config. See screen capture below.
![ff3](https://user-images.githubusercontent.com/487038/36978027-77b7f71e-2083-11e8-90b1-61c17b7a097b.png)

**Comment by JC5** [comment]
@Nik-vr how did you download the new release exactly? Your files seem to be corrupted.

@jinformatique a fix is coming.

**Comment by JC5** [comment]
I've just released 4.7.1.4. The issue should be fixed.

@Nik-vr please open a new issue if your problems persist.

@jinformatique likewise. Make sure to clear your cache.

**Comment by luangrezende** [comment]
did u solved it?
